### ISSUE TITLE
HttpRequestException: 400 (Bad Request)

### ISSUE THREAD

**Issue by maximvhs** [reown-com/reown-dotnet#33]
Hi, I've been trying to add this testnet to the example app

https://docs.onbeam.com/chain/get-started/rpc#beam-testnet-fuji

Here is my config
```
// AppKit configuration
var appKitConfig = new AppKitConfig
{
    // Project ID from https://cloud.reown.com/
    projectId = "884a108399b5e7c9bc00bd9be4ccb2cc",
    metadata = new Metadata(
        "AppKit Unity",
        "AppKit Unity Sample",
        "https://reown.com",
        "https://raw.githubusercontent.com/reown-com/reown-dotnet/main/media/appkit-icon.png",
        new RedirectData
        {
            // Used by native wallets to redirect back to the app after approving requests
            Native = "appkit-sample-unity://"
        }
    ),
    customWallets = GetCustomWallets(),
    // On mobile show 5 wallets on the Connect view (the first AppKit modal screen)
    connectViewWalletsCountMobile = 5,
    // Assign the SIWE configuration created above. Can be null if SIWE is not used.
    siweConfig = siweConfig,
    supportedChains = new Chain[]
    {
        new(ChainConstants.Namespaces.Evm,
            chainReference: "13337",
            name: "Beam Testnet",
            nativeCurrency: new Currency("Beam", "BEAM", 18),
            blockExplorer: new BlockExplorer("Beam Subnet Block Explorer", "https://subnets-test.avax.network/beam"),
            rpcUrl: "https://build.onbeam.com/rpc/testnet",
            isTestnet: true,
            imageUrl: "https://images.ctfassets.net/gcj8jwzm6086/2ZXZw0POSuXhwoGTiv2fzh/5b9d9e81acb434461da5addb1965f59d/chain-logo.png")
    }
};
```
When I try to GetBalance an error occures
```
[21:23:58][127,8s at #7507]: HttpRequestException: 400 (Bad Request)
System.Net.Http.HttpResponseMessage.EnsureSuccessStatusCode () (at <6e0ff4b8016b42cc9117913bee5de47c>:0)
Nethereum.JsonRpc.Client.RpcClient.SendAsync (Nethereum.JsonRpc.Client.RpcMessages.RpcRequestMessage request, System.String route) (at <cc738ddfb9674046a771b077a90ea9aa>:0)
Rethrow as RpcClientUnknownException: Error occurred when trying to send rpc requests(s): eth_getBalance
Nethereum.JsonRpc.Client.RpcClient.SendAsync (Nethereum.JsonRpc.Client.RpcMessages.RpcRequestMessage request, System.String route) (at <cc738ddfb9674046a771b077a90ea9aa>:0)
Nethereum.JsonRpc.Client.ClientBase.SendInnerRequestAsync[T] (Nethereum.JsonRpc.Client.RpcMessages.RpcRequestMessage reqMsg, System.String route) (at <0a51c1be044b405187968f01e268c47d>:0)
Nethereum.JsonRpc.Client.RequestInterceptor.InterceptSendRequestAsync[T] (System.Func`3[T1,T2,TResult] interceptedSendRequestAsync, Nethereum.JsonRpc.Client.RpcRequest request, System.String route) (at <0a51c1be044b405187968f01e268c47d>:0)
Reown.Sign.Nethereum.ReownInterceptor.InterceptSendRequestAsync[T] (System.Func`3[T1,T2,TResult] interceptedSendRequestAsync, Nethereum.JsonRpc.Client.RpcRequest request, System.String route) (at <8c0e02d589664788ab58a3105666849c>:0)
Nethereum.JsonRpc.Client.ClientBase.SendRequestAsync[T] (Nethereum.JsonRpc.Client.RpcRequest request, System.String route) (at <0a51c1be044b405187968f01e268c47d>:0)
Reown.AppKit.Unity.NethereumEvmService.GetBalanceAsyncCore (System.String address) (at <33c188ddddc8469b90fbcf1e2b72c70d>:0)
Sample.Dapp.OnGetBalanceButton () (at <c59fc2910f9f4a31b2b0a6ac7ce9549c>:0)
UnityEngine.Debug:LogException(Exception, Object)
Sample.<OnGetBalanceButton>d__10:MoveNext()
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1:SetException(Exception)
Reown.AppKit.Unity.<GetBalanceAsyncCore>d__12:MoveNext()
UnityEngine.UnitySynchronizationContext:ExecuteTasks()
```

**Comment by skibitsky** [comment]
Hello,

Reown's Blockchain API does not currently support the Beam chain. You can view the complete list of supported chains [here](https://github.com/reown-com/blockchain-api/blob/master/SUPPORTED_CHAINS.md).

In this case, AppKit should switch to the RPC URL you provided in the `Chain` object (`https://build.onbeam.com/rpc/testnet`), but for some reason, it isn’t doing so. I'll investigate and get back to you.

**Comment by maximvhs** [comment]
Thank you @skibitsky,

I want to migrate to the Reown version.

I have a wokring setup with this chain using deprecated WalletConnectSharp and Nethereum. I just create a Web3 instance and provide an implementation of IAccount that uses SignClient to send transactions. Maybe this will be useful.

```
SessionStruct session; //populated as a result of established connection with the wallet
.
.
.
var signClientWithSession = new SignClientWithSession()
{
    topic = session.Topic,
    accountAddress = accountAddress,
    chainId = int.Parse(chainId),
    signClient = walletConnectUnity.SignClient,
};

var chain = Chain.chains[signClientWithSession.chainId];

var rpcClient = new RpcClient(new Uri(chain.rpcAddress));
var web3 = new Web3(new WCWeb3Account(signClientWithSession, rpcClient), rpcClient);
var actionProvider = new WCActionProvider(web3, chain, accountAddress);
account = new WCEthAccount(accountAddress, session.Topic, walletConnectUnity.SignClient, actionProvider);
.
.
.
// inside Action Provider
var ethBalance = await web3.Eth.GetBalance.SendRequestAsync(publicAddress);
```

```
public class SignClientWithSession
{
    public string topic;
    public string accountAddress;

    public int chainId;
    public ISignClient signClient;
}
.
.
.
public class WCWeb3Account : IAccount
{
    public string Address { get; private set; }

    public ITransactionManager TransactionManager { get; private set; }

    public INonceService NonceService { get; set; }

    public IAccountSigningService AccountSigningService { get; private set; }

    public WCWeb3Account(SignClientWithSession signClientWithSession, IClient client)
    {
        Address = signClientWithSession.accountAddress;
        NonceService = new InMemoryNonceService(Address, client);
        TransactionManager = new WCWeb3TransactionManager(client, signClientWithSession, NonceService);
    }
}
.
.
.
public class WCWeb3TransactionManager : TransactionManager
{
    readonly StringBuilder debugString = new();
    readonly SignClientWithSession signClientWithSession;
    readonly INonceService nonceService;

    public WCWeb3TransactionManager(IClient client, SignClientWithSession signClientWithSession, INonceService nonceService) : base(client)
    {
        this.signClientWithSession = signClientWithSession;
        this.nonceService = nonceService;
    }

    public override Task<string> SignTransactionAsync(TransactionInput transaction)
    {
        debugString.Clear();
        debugString.AppendLine("---- Custom SIGN transaction called ----");
        debugString.AppendLine("Transaction:");
        debugString.AppendLine($"  From: {transaction.From}");
        debugString.AppendLine($"  To: {transaction.To}");
        debugString.AppendLine($"  Value: {transaction.Value}");
        debugString.AppendLine($"  Gas: {transaction.Gas}");
        debugString.AppendLine($"  GasPrice: {transaction.GasPrice}");
        Debug.Log(debugString);

        return base.SignTransactionAsync(transaction);
    }

    public override async Task<string> SendTransactionAsync(TransactionInput transactionInput)
    {
        if (transactionInput.ChainId == null)
        {
            transactionInput.ChainId = new HexBigInteger(new BigInteger(signClientWithSession.chainId));
        }

        if (transactionInput.Nonce == null)
        {
            var nextNonce = await nonceService.GetNextNonceAsync();
            transactionInput.Nonce = nextNonce;
        }

        if (transactionInput.Type == null)
        {
            transactionInput.Type = new HexBigInteger(new BigInteger(TransactionType.EIP1559.AsByte()));
        }

        if (transactionInput.Gas == null)
        {
            var estimatedGas = await EstimateGasAsync(transactionInput);
            transactionInput.Gas = estimatedGas;
        }

        await SetTransactionFeesOrPricingAsync(transactionInput);

        var transaction = new EthSendTransaction(new Transaction()
        {
            from = transactionInput.From,
            To = transactionInput.To,
            Value = transactionInput.Value.ToString(),
            Gas = transactionInput.Gas.ToString(),
            Data = transactionInput.Data,
        });

        debugString.Clear();
        debugString.AppendLine("---- Custom SEND transaction called ----");
        debugString.AppendLine("Transaction:");
        debugString.AppendLine($"  From: {transactionInput.From}");
        debugString.AppendLine($"  To: {transactionInput.To}");
        debugString.AppendLine($"  Value: {transactionInput.Value}");
        debugString.AppendLine($"  Gas: {transactionInput.Gas}");
        debugString.AppendLine($"  GasPrice: {transactionInput.GasPrice}");
        Debug.Log(debugString);

        await UniTask.SwitchToMainThread();

        return await signClientWithSession.signClient.Request<EthSendTransaction, string>(signClientWithSession.topic, transaction);
    }
}
```

**Comment by skibitsky** [comment]
Fixed in 1.2.0
